lib/loadGltfUris.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var isDataUri = require('./isDataUri');
+var dataUriToBuffer = require('data-uri-to-buffer');
+var async = require('async');
+
+module.exports = loadGltfUris;
+
+function loadGltfUris(gltf, basePath, callback) {
+    var shaders = gltf.shaders;
+
+    //Iterate through each shader and load its uri
+    if (defined(shaders)) {
+        var shaderIds = Object.keys(shaders);
+        async.each(shaderIds, function(shaderId, asyncCallback) {
```

Comment:
Why exactly is this code using `async.each`?  Why not go through each property in `shaders` as we did for the removal optimizations?  I wouldn't bother adding parallelism yet as this might not even be the right granularity.